filechecker.py

```diff
@@ -8,7 +8,10 @@ def parse_arguments():
 
 def checkpath(path):
     if os.path.exists(path):
-        print("Valid path!")
+        with open(path) as reader:
+            char = reader.read()
+            print(char)
+            print(len(char))
```

Comment:
What happens if the path is a directory? <-- tricky question :D

Reply:
In that case, both "if os.path.exists(path):" and "elif os.path.isdir(path):" are true, which first causes an error, as it never gets to the part of the code that is supposed to catch it before it gets to that point. We can fix this by reversing the order of operations, making the code check if its a directory first